pkg/mempool/simplemempool/internal/parts/computeids/computeids.go

```diff
@@ -19,13 +19,27 @@ import (
 func IncludeComputationOfTransactionAndBatchIDs(
 	m dsl.Module,
 	mc common.ModuleConfig,
-	_ *common.ModuleParams,
+	params *common.ModuleParams,
 	_ *common.State,
 ) {
 	mppbdsl.UponRequestTransactionIDs(m, func(txs []*trantorpbtypes.Transaction, origin *mppbtypes.RequestTransactionIDsOrigin) error {
-		txMsgs := make([]*hasherpbtypes.HashData, len(txs))
-		for i, tx := range txs {
-			txMsgs[i] = &hasherpbtypes.HashData{Data: serializeTXForHash(tx.Pb())}
+		if txs == nil || len(txs) > params.MaxTransactionsInBatch {
+			// Invalid request, ignore
+			return nil
+		}
+
+		txMsgs := make([]*hasherpbtypes.HashData, 0, len(txs))
+		for _, tx := range txs {
+			serializedTx := serializeTXForHash(tx.Pb())
+			if serializedTx == nil {
+				continue
+			}
+			txMsgs = append(txMsgs, &hasherpbtypes.HashData{Data: serializedTx})
+		}
+
+		// There should be at least one transaction, otherwise we would not have gotten here (see batchfetcher/UponDeliverCert)
```

Comment:
Is this a sanity check? If yes, we should return an error if it fails.

Reply:
My understanding is that returning an error in a dsl.Upon function halts execution, and we do not want that if a corrupted node sends us a corrupted batch with empty transactions.

Reply:
Either way, just returning nil in the above conservation (i.e. as soon as a tx is nil) and checking with the check at the beginning that txs != nil would suffice. Done this.

Reply:
Same point as above. If it can happen at [UponProvideBatchMessageReceived](https://github.com/filecoin-project/mir/blob/8f352300a01bb71d1bc05cb62a92e0e105ad0862/pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go#L128-L142), catch it there. I think it's good practice to sanitize input as close to the input itself as possible, not deeper down the code path. Things getting silently ignored somewhere deep in the code is the best recipe for debugging nightmares.

Reply:
Done for both [UponProvideBatchMessageReceived](https://github.com/filecoin-project/mir/blob/8f352300a01bb71d1bc05cb62a92e0e105ad0862/pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go#L128-L142) and UponRequestSigMessageReceived (see commit for this). As mentioned, the upper bound check is for the moment still in mempool.